tests/run-pass/ptr_int_oob.rs

```diff
@@ -0,0 +1,48 @@
+// compile-flags: -Zmiri-disable-stacked-borrows
```

Comment:
Could you rename the file to `ptr_int_roundtrip` or so?

Please also add a test that doesn't do a "perfect" roundtrip, e.g. for an array of `u8`, cast a ptr to `array[0]` to int, then add 2, and then cast that back and access it.